wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireTask.kt

```diff
@@ -34,10 +35,10 @@ open class WireTask : SourceTask() {
   var pluginVersion: String = VERSION
 
   @get:Internal
-  internal lateinit var sourceInput: WireInput
+  internal lateinit var sourceInput: List<Location>
```

Comment:
This is the main change: WireInput cannot be serialized during instant execution (a.k.a config caching) because it has a configuration which is not serializable.
The workaround is to use the ` Locations` data class which seem to be handled well by Gradle. I faced an another bug that prevented me to make sure it works 100%.